python-client/giskard/ml_worker/testing/registry/giskard_test.py

```diff
@@ -129,7 +129,13 @@ def __call__(self, *args, **kwargs) -> "GiskardTestMethod":
     def execute(self) -> Result:
         analytics.track("test:execute", {"test_name": self.meta.full_name})
 
-        return self.test_fn(**self.params)
+        try:
+            return self.test_fn(**self.params)
```

Comment:
you can inspect the `self.test_fn` and see if it has `debug` argument instead of trying to run it and catching and exception